src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -183,13 +189,84 @@ bool isDefinedByONNXConstantOp(Value v) {
   return isa<ONNXConstantOp>(v.getDefiningOp());
 }
 
+bool LegalExpandPowOpToMul(ONNXPowOp op) {
+  Value exponent = op.Y();
+  if (!isDefinedByONNXConstantOp(exponent))
+    return true;
+
+  if (auto constOp = dyn_cast<ONNXConstantOp>(exponent.getDefiningOp())) {
+    if (DenseElementsAttr dataAttr =
+            constOp.valueAttr().dyn_cast<DenseElementsAttr>()) {
+      if (dataAttr.getNumElements() == 1) {
+        Type elementType = dataAttr.getElementType();
+        if (elementType.isa<FloatType>()) {
+          auto valueIt = dataAttr.getValues<APFloat>().begin();
+          double val = (*valueIt).convertToDouble();
```

Comment:
Val >= 0?

Reply:
Good catch! Thanks! I added `val >= 0`.

Reply:
Even though you can use multiple for val == 0 in math, your implementation does not start with initial value 1.
```
Result = constant 1
for (I = 0; I <val; I++) {
    result = multiple(result, x);
}
```
Or you can check val > 0 to keep the current code.

Reply:
Fixed it.  Thanks!